dotCMS/src/main/java/com/dotcms/vanityurl/action/ViewVanityURLAction.java

```diff
@@ -0,0 +1,26 @@
+package com.dotcms.vanityurl.action;
+
+import com.dotcms.repackage.javax.portlet.PortletConfig;
+import com.dotcms.repackage.javax.portlet.RenderRequest;
+import com.dotcms.repackage.javax.portlet.RenderResponse;
+import com.dotcms.repackage.javax.portlet.WindowState;
+import com.dotcms.repackage.org.apache.struts.action.ActionForm;
+import com.dotcms.repackage.org.apache.struts.action.ActionForward;
+import com.dotcms.repackage.org.apache.struts.action.ActionMapping;
+import com.dotmarketing.portal.struts.DotPortletAction;
+
+public class ViewVanityURLAction extends DotPortletAction {
+	
+	public ActionForward render(ActionMapping mapping, ActionForm form,
+			PortletConfig config, RenderRequest req, RenderResponse res)
+	throws Exception {
+		
+		if (req.getWindowState().equals(WindowState.NORMAL)) {
+			return mapping.findForward("portlet.ext.virtuallinks.view");
+		} else {			
+			return mapping.findForward("portlet.ext.virtuallinks.view_virtuallinks");
+
+		}
```

Comment:
I prefer to reduce it to

return mapping.findForward (req.getWindowState().equals(WindowState.NORMAL)?
   "portlet.ext.virtuallinks.view":"portlet.ext.virtuallinks.view_virtuallinks");

Also they should be a constants